src/app/features/issue/providers/gitlab/gitlab-common-interfaces.service.ts

```diff
@@ -111,88 +111,108 @@ export class GitlabCommonInterfacesService implements IssueServiceInterface {
 
     const wasUpdated = lastRemoteUpdate > (task.issueLastUpdated || 0);
 
-    if (wasUpdated) {
+    if (wasUpdated || idFormatChanged) {
       return {
         taskChanges: {
           ...this.getAddTaskData(issue),
           issueWasUpdated: true,
         },
         issue,
-        issueTitle: this._formatIssueTitleForSnack(issue.number, issue.title),
+        issueTitle: this._formatIssueTitleForSnack(issue),
       };
     }
     return null;
   }
 
   async getFreshDataForIssueTasks(
     tasks: Task[],
-  ): Promise<{ task: Task; taskChanges: Partial<Task>; issue: GitlabIssue }[]> {
-    // First sort the tasks by the issueId
-    // because the API returns it in a desc order by issue iid(issueId)
-    // so it makes the update check easier and faster
-    tasks.sort((a, b) => +(b.issueId as string) - +(a.issueId as string));
+  ): Promise<{ task: Task; taskChanges: Partial<Task>; issue: GitlabIssue | null }[]> {
     const projectId = tasks && tasks[0].projectId ? tasks[0].projectId : 0;
     if (!projectId) {
       throw new Error('No projectId');
     }
 
     const cfg = await this._getCfgOnce$(projectId).toPromise();
-    const issues: GitlabIssue[] = [];
+    const issues = new Map<string, GitlabIssue>();
     const paramsCount = 59; // Can't send more than 59 issue id For some reason it returns 502 bad gateway
-    let ids;
+    const iidsByProject = new Map<string, string[]>();
     let i = 0;
-    while (i < tasks.length) {
-      ids = [];
-      for (let j = 0; j < paramsCount && i < tasks.length; j++, i++) {
-        ids.push(tasks[i].issueId);
+
+    for (const task of tasks) {
+      if (!task.issueId) {
+        continue;
+      }
+      const project = this._gitlabApiService.getProject$(cfg, task.issueId);
+      if (!iidsByProject.has(project)) {
+        iidsByProject.set(project, []);
       }
-      issues.push(
-        ...(await this._gitlabApiService.getByIds$(ids as string[], cfg).toPromise()),
-      );
+      iidsByProject.get(project)?.push(task.issueId as string);
     }
 
+    iidsByProject.forEach(async (allIds, project) => {
+      for (i = 0; i < allIds.length; i += paramsCount) {
+        (
+          await this._gitlabApiService
```

Comment:
Using Promise.all is probably much faster since the requests can be handled in parallel.